smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/HTTPBindingProtocolGenerator.kt

```diff
@@ -415,16 +415,12 @@ abstract class HTTPBindingProtocolGenerator(
             operationMiddleware.appendMiddleware(operation, RetryMiddleware(ctx.model, ctx.symbolProvider, retryErrorInfoProviderSymbol))
             operationMiddleware.appendMiddleware(operation, SignerMiddleware(ctx.model, ctx.symbolProvider))
             addProtocolSpecificMiddleware(ctx, operation)
-            /*
-             * Auth scheme middleware must be appended to codegen operation stack AFTER protocol specific middleware is appended.
-             * This is because endpoint middleware must be generated into client operation stack first with position: .before,
-             * AFTER which auth scheme must be generated into client operation stack with position: .before, which ensures
-             * auth scheme middleware is executed FIRST before endpoint middleware is executed, as SRA flow defines.
-             */
             operationMiddleware.appendMiddleware(operation, AuthSchemeMiddleware(ctx.model, ctx.symbolProvider))
             for (integration in ctx.integrations) {
                 integration.customizeMiddleware(ctx, operation, operationMiddleware)
             }
+            // must be last to support adding business metrics
+            addUserAgentMiddleware(ctx, operation)
```

Comment:
bug fix: addUserAgentMiddleware handles feature id (business metric) tracking. This was supposed to be last in the stack but it never was prior to this change due to protocol-specific middleware